stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_connector.py

```diff
@@ -63,22 +64,30 @@ def create_results_connection(self, query, offset, length):
         :param query: str, search_id
         :param offset: int,offset value
         :param length: int,length value"""
-
+        response = None
         response_dict = dict()
         return_obj = dict()
         length = int(length)
+        offset = int(offset)
+
+        # total records is the sum of the offset and length(limit) value
+        total_records = offset + length
 
         try:
             if self.init_error:
                 print("Token Generation Failed:")
                 return self.adal_response
-            response = self.api_client.run_search(query, offset, length)
+            # check for length value against the max limit(1000) of $top param in data source
+            if length <= self.max_limit:
+                response = self.api_client.run_search(query, offset + length)
```

Comment:
`offset + length` -> `total_records`

Reply:
@mdazam1942 This change is done and code is updated.